File: app/src/main/java/com/apps/straightface/gymstat/Data/ExerciseEntry.java
package com.apps.straightface.gymstat.Data;

import com.apps.straightface.gymstat.Utils.WeightUnitRealm;

import java.util.Date;

import io.realm.RealmObject;

public class ExerciseEntry extends RealmObject
{
    private Date date;
    private int setNumber;
    private double weight;
    private WeightUnitRealm unit;
    private int numberOfReps;
    private boolean bestSet;

    public Date getDate()
    {
        return date;
    }
    public void setDate(Date date)
    {
        this.date = date;
    }

    public int getSetNumber()
    {
        return setNumber;
    }
    public void setSetNumber(int setNumber)
    {
        this.setNumber = setNumber;
    }

    public double getWeight()
    {
        return weight;
    }
    public void setWeight(double weight)
    {
        this.weight = weight;
    }

    public WeightUnitRealm getUnit()
    {
        return unit;
    }
    public void setUnit(WeightUnitRealm unit)
    {
        this.unit = unit;
    }

    public int getNumberOfReps()
    {
        return numberOfReps;
    }
    public void setNumberOfReps(int numberOfReps)
    {
        this.numberOfReps = numberOfReps;
    }

    public boolean isBestSet()
    {
        return bestSet;
    }
    public void setBestSet(boolean bestSet)
    {
        this.bestSet = bestSet;
    }
}

File: app/src/main/java/com/jorbital/gymstat/data/ExerciseEntry.kt
package com.jorbital.gymstat.data

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import java.util.*

open class ExerciseEntry : RealmObject() {
    @PrimaryKey
    var idKey: String? = null
    var date: Date? = null
    var setNumber: Int = 0
    var weight: Double = 0.toDouble()
    var unit: WeightUnitRealm? = null
    var numberOfReps: Int = 0
    var isBestSet: Boolean = false
}
